src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobWorker.cs

```diff
@@ -50,30 +53,33 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             {
                 try
                 {
-                    // Remove all completed tasks.
-                    runningTasks.RemoveAll(task => task.IsCompleted);
-
-                    // Get list of available jobs.
-                    if (runningTasks.Count < _exportJobConfiguration.MaximumNumberOfConcurrentJobsAllowed)
+                    if (_storageReady)
```

Comment:
nit: wondering if we can use this check in while statement itself instead of adding one more if block for better code readability?
probably while (!cancellationToken.IsCancellationRequested && _storageReady)

Reply:
There is a delayed wait time logic inside the while loop, but pulled out an "if" statement from the try block